addons/sms/tests/common.py

```diff
@@ -255,20 +271,25 @@ def assertSMSNotification(self, recipients_info, content, messages=None, check_s
             self.assertEqual(notif.author_id, notif.mail_message_id.author_id, 'SMS: Message and notification should have the same author')
             for field_name, expected_value in (mail_message_values or {}).items():
                 self.assertEqual(notif.mail_message_id[field_name], expected_value)
+            if 'failure_reason' in recipient_info:
+                self.assertEqual(notif.failure_reason, recipient_info['failure_reason'])
             if state not in {'process', 'sent', 'ready', 'canceled', 'pending'}:
                 self.assertEqual(notif.failure_type, recipient_info['failure_type'])
-            if check_sms:
+
+            recipient_check_sms = recipient_info.get('recipient_check_sms', check_sms)
+            if recipient_check_sms:
```

Comment:
I'd argue this is not less readable
```suggestion
            if recipient_info.get('recipient_check_sms', check_sms):
```